### PR TITLE
feat(auth): aprimora página 403 de acesso negado

### LINKED ISSUE #75: [FE] Criar página de acesso negado 403
## Descrição do problema / necessidade

O frontend precisa tratar de forma amigável os casos em que um usuário autenticado tenta acessar uma funcionalidade para a qual não possui permissão.

Atualmente o sistema já trata autenticação e redirecionamento em caso de `401`, mas ainda é necessário criar uma experiência específica para erro `403 Forbidden`, especialmente para rotas administrativas protegidas por role.

Sem essa página, o usuário pode ser redirecionado de forma genérica ou ficar sem uma explicação clara sobre o motivo do bloqueio.

## Objetivo

Criar uma página de acesso negado para tratar cenários de autorização insuficiente, exibindo uma mensagem clara e permitindo que o usuário retorne para uma área segura do sistema.

## Escopo da implementação

- Criar componente de acesso negado, por exemplo:

```text
src/app/pages/errors/access-denied/
```

ou

```text
src/app/pages/acesso-negado/
```

- Criar rota pública ou autenticada:

```text
/acesso-negado
```

- Exibir mensagem amigável informando que o usuário não possui permissão para acessar a funcionalidade.
- Adicionar botão/link para voltar ao dashboard ou página inicial.
- Ajustar o guard por role para redirecionar usuários sem permissão para essa página.
- Ajustar interceptor HTTP para tratar respostas `403 Forbidden`, quando fizer sentido.
- Manter tratamento de `401 Unauthorized` separado do tratamento de `403 Forbidden`.

## Critérios de aceite

- [ ] Existe rota `/acesso-negado`.
- [ ] Existe componente visual para acesso negado.
- [ ] A página exibe mensagem clara e amigável.
- [ ] A página possui ação para voltar ao dashboard ou página inicial.
- [ ] Usuário autenticado sem permissão é redirecionado para `/acesso-negado` pelo guard de role.
- [ ] Respostas `403` da API são tratadas sem quebrar a aplicação.
- [ ] O tratamento de `401` continua redirecionando para login.
- [ ] O tratamento de `403` não remove o token do usuário, salvo decisão contrária explícita.
- [ ] Testes unitários cobrem o componente e o redirecionamento esperado.

### Testes unitários

- Deve renderizar a mensagem de acesso negado.
- Deve possuir link ou botão para voltar ao dashboard.
- O guard por role deve redirecionar para `/acesso-negado` quando usuário não possuir permissão.
- O interceptor deve tratar `403` sem executar logout automático, caso essa seja a regra definida.
- O interceptor deve continuar tratando `401` como sessão inválida/expirada.

### Validação manual sugerida

1. Fazer login com usuário `USER`.
2. Tentar acessar uma rota administrativa diretamente pela URL.
3. Validar que o usuário é redirecionado para `/acesso-negado`.
4. Validar que a mensagem exibida é clara.
5. Clicar no botão de voltar ao dashboard.
6. Validar que o usuário continua autenticado.
7. Simular uma resposta `403` da API em uma funcionalidade protegida.
8. Validar que a aplicação não quebra e exibe tratamento adequado.
9. Simular uma resposta `401` e validar que o comportamento de logout/redirecionamento para login continua funcionando.

## Observações técnicas

- Esta issue complementa a criação do guard por role.
- `401 Unauthorized` e `403 Forbidden` devem ter tratamentos diferentes:
  - `401`: usuário não autenticado, token ausente, inválido ou expirado;
  - `403`: usuário autenticado, mas sem permissão suficiente.
- Evitar fazer logout automático em `403`, pois o usuário pode estar autenticado corretamente, apenas sem permissão para aquela ação.
- A página deve seguir o padrão visual atual do sistema.
- A mensagem deve ser simples, por exemplo:

```text
Acesso negado. Você não possui permissão para acessar esta funcionalidade.
```

- Considerar futuramente uma página genérica para erros `404`, `403` e outros estados, mas esta issue deve focar apenas no `403`.

## Labels sugeridas

- `frontend`
- `security`
- `access-control`
- `ui`
- `enhancement`

### PR DESCRIPTION
**Pull request by FabioCarlesso**
### Motivation
- Fornecer uma página 403 clara e acessível para usuários autenticados que tentam acessar áreas sem permissão (issue #75).

### Description
- Reescreve o componente de `ForbiddenComponent` para usar marcação semântica e acessível e adiciona ação secundária `voltar()` que chama `Location.back()` no componente (`src/app/pages/auth/forbidden/forbidden.component.ts`).
- Substitui o conteúdo da view por texto orientativo, código 403 em destaque e ações primária/secundária (`Ir para o início` e `Voltar`) em `forbidden.component.html`.
- Atualiza estilos para seguir os tokens do Design System e melhorar responsividade e layout (`forbidden.component.scss`).
- Atualiza os testes unitários do componente para cobrir título, mensagem, link principal e clique da ação secundária (`forbidden.component.spec.ts`).

### Testing
- Executed `npm test -- --watch=false --browsers=ChromeHeadless`, which compiled the tests but failed to launch `ChromeHeadless` due to missing `CHROME_BIN` in the environment (Karma could not start the browser).
- Executed `npm run build`, which failed in production mode due to external font inlining returning HTTP 403 (environment limitation), so the build could not complete in production configuration.
- Executed `npm run build -- --configuration development`, which completed successfully and produced application bundles.

Files changed: `src/app/pages/auth/forbidden/forbidden.component.ts`, `src/app/pages/auth/forbidden/forbidden.component.html`, `src/app/pages/auth/forbidden/forbidden.component.scss`, `src/app/pages/auth/forbidden/forbidden.component.spec.ts`.

Closes #75

------
[Codex Task](https://chatgpt.com/codex/cloud/tasks/task_e_69fd546404848328b021b2e852de084e)

### CODE REVIEW

File: src/app/pages/auth/forbidden/forbidden.component.ts [hunk 1]
```diff
@@ -8,4 +9,10 @@ import { RouterLink } from '@angular/router';
   templateUrl: './forbidden.component.html',
   styleUrl: './forbidden.component.scss'
 })
-export class ForbiddenComponent {}
+export class ForbiddenComponent {
+  private readonly location = inject(Location);
+
+  voltar(): void {
+    this.location.back();
```
Comment (FabioCarlesso) [inline src/app/pages/auth/forbidden/forbidden.component.ts]: **🔴 Bug – Redirect loop with `Location.back()` (Severity: High)**

- **Problem:** `Location.back()` replays the previous browser history entry. When the user is redirected to `/403` by `roleGuard`, the previous entry *is the restricted route itself* (e.g. `/profissionais`). Clicking "Voltar" pushes the user back there, `roleGuard` fires again, and they're redirected to `/403` once more — an infinite bounce loop.
- **Why it matters:** The "Voltar" button becomes actively harmful for the most common use case: a user who tried to access a route they're not authorised for.
- **Suggested improvement:** Guard against navigating back to a restricted page. One safe approach is to check whether the router's current navigation extras carry a `redirectedFrom` hint, or simply always fall back to `/` when there is no safe prior route:

```typescript
voltar(): void {
  // Location.back() would loop back to the forbidden page in the common
  // redirect-from-roleGuard scenario, so fall back to home when the
  // referrer is the same origin (i.e. an in-app redirect).
  const referrer = document.referrer;
  if (referrer && new URL(referrer).origin !== window.location.origin) {
    this.location.back();
  } else {
    this.router.navigate(['/']);
  }
}
```
Or, more simply: replace "Voltar" with a well-labelled secondary destination (e.g. "Ir para o Dashboard") so the user always has a safe exit, without relying on history at all.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/app/pages/auth/forbidden/forbidden.component.html [hunk 2]
```diff
@@ -1,8 +1,17 @@
...
+
+    <h1 id="forbidden-title" class="forbidden-title">Acesso negado</h1>
+    <p class="forbidden-message">
+      Seu perfil não possui permissão para acessar esta área do sistema.
+      Caso precise liberar o acesso, entre em contato com um administrador.
+    </p>
+
+    <div class="forbidden-actions" aria-label="Ações para sair da página de acesso negado">
+      <a routerLink="/" class="btn btn-primary">Ir para o início</a>
+      <button type="button" class="btn btn-secondary" (click)="voltar()">Voltar</button>
```
Comment (FabioCarlesso) [inline src/app/pages/auth/forbidden/forbidden.component.html:13]: **🟡 Bug – `Location.back()` with no navigation history (Severity: Medium)**

- **Problem:** If the user opens `/403` directly (bookmark, typed URL, email link, or fresh tab from an external redirect), `window.history` has no previous entry. `Location.back()` will either do nothing or navigate the browser out of the app entirely to whatever site was open before.
- **Why it matters:** Users landing on this page from outside the app will have a "Voltar" button that silently fails or ejects them from the application.
- **Suggested improvement:** Disable the button when there is no safe history to go back to, or replace it with an explicit secondary destination (e.g. "Ir para o Dashboard") that always works regardless of entry point.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/app/pages/auth/forbidden/forbidden.component.html [hunk 3]
```diff
@@ -1,8 +1,17 @@
-<div class="forbidden-container">
-  <div class="forbidden-card">
-    <h1 class="forbidden-code">403</h1>
-    <p class="forbidden-title">Acesso não autorizado</p>
-    <p class="forbidden-message">Você não tem permissão para acessar esta página.</p>
-    <a routerLink="/" class="btn-home">Voltar ao início</a>
-  </div>
-</div>
+<section class="forbidden-container" aria-labelledby="forbidden-title">
+  <article class="forbidden-card">
```
Comment (FabioCarlesso) [inline src/app/pages/auth/forbidden/forbidden.component.html:2]: **🔵 Semantics – Incorrect use of `<article>` (Severity: Low)**

- **Problem:** `<article>` is a sectioning element intended for self-contained content that would make sense if distributed independently (e.g. a blog post, news article, forum reply). An error-page card does not meet that definition.
- **Why it matters:** Screen readers and assistive technology announce landmark roles; an incorrect landmark can confuse users navigating by region. It also goes against WHATWG semantics.
- **Suggested improvement:** Replace `<article>` with a plain `<div>` (since the outer `<section>` already provides the landmark boundary) or keep it as a second `<section>`:

```html
<section class="forbidden-container" aria-labelledby="forbidden-title">
  <div class="forbidden-card">
    ...
  </div>
</section>
```

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/app/pages/auth/forbidden/forbidden.component.spec.ts [hunk 4]
```diff
@@ -1,11 +1,17 @@
+import { Location } from '@angular/common';
 import { TestBed } from '@angular/core/testing';
 import { RouterTestingModule } from '@angular/router/testing';
```
Comment (FabioCarlesso) [inline src/app/pages/auth/forbidden/forbidden.component.spec.ts]: **🔵 Angular Best Practices – `RouterTestingModule` is deprecated (Severity: Low–Medium)**

- **Problem:** `RouterTestingModule` was deprecated in Angular 16 and is not the recommended approach in Angular 19 standalone projects.
- **Why it matters:** Using deprecated APIs will eventually break when they are removed, and it deviates from the project's modern standalone approach.
- **Suggested improvement:** Replace with `provideRouter([])` from `@angular/router`:

```typescript
import { provideRouter } from '@angular/router';

await TestBed.configureTestingModule({
  imports: [ForbiddenComponent],
  providers: [
    provideRouter([]),
    { provide: Location, useValue: locationSpy }
  ]
}).compileComponents();
```

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/app/pages/auth/forbidden/forbidden.component.scss [hunk 5]
```diff
@@ -2,54 +2,78 @@
   display: flex;
   align-items: center;
   justify-content: center;
-  min-height: 100vh;
-  background: var(--bg-app);
+  min-height: calc(100vh - 64px - (var(--gutter) * 2));
```
Comment (FabioCarlesso) [inline src/app/pages/auth/forbidden/forbidden.component.scss:5]: **🔵 UI/UX – Magic number hardcodes navbar height (Severity: Low)**

- **Problem:** `64px` is the current navbar height hardcoded directly in the calculation. If the navbar height ever changes (responsive breakpoint, density setting, redesign), this value will silently break the centering.
- **Why it matters:** The Design System already uses tokens for density-sensitive sizes (`--btn-h`, `--row-h`, etc.). Hardcoded pixel values resist refactoring.
- **Suggested improvement:** Introduce a `--navbar-h` token in `_tokens.scss` (or reference an existing one if it already exists) and use it here:

```scss
min-height: calc(100vh - var(--navbar-h) - (var(--gutter) * 2));
```

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/app/pages/auth/forbidden/forbidden.component.spec.ts [hunk 6]
```diff
@@ -16,26 +22,42 @@ describe('ForbiddenComponent', () => {
     expect(fixture.componentInstance).toBeTruthy();
   });
 
-  it('should display 403 code', () => {
+  it('should display 403 code and access denied title', () => {
```
Comment (FabioCarlesso) [inline src/app/pages/auth/forbidden/forbidden.component.spec.ts:28]: **🔵 Tests – Fixture not destroyed between tests (Severity: Low)**

- **Problem:** Each `it()` block creates its own `ComponentFixture` via `TestBed.createComponent(ForbiddenComponent)` but never calls `fixture.destroy()`. This leaves live DOM nodes attached between tests and can cause cross-test interference in larger suites.
- **Why it matters:** In this small suite it rarely causes failures, but it is an antipattern that accumulates tech debt.
- **Suggested improvement:** Move fixture creation to `beforeEach` and add `afterEach(() => fixture.destroy())`:

```typescript
let fixture: ComponentFixture<ForbiddenComponent>;

beforeEach(() => {
  fixture = TestBed.createComponent(ForbiddenComponent);
  fixture.detectChanges();
});

afterEach(() => fixture.destroy());
```

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by FabioCarlesso — COMMENTED** [review]
## Code Review — PR #82: feat(auth): aprimora página 403 de acesso negado

Good effort overall. The accessibility improvements (semantic landmark, `aria-labelledby`, `aria-hidden` on the decorative number), the use of Design System tokens in SCSS, the modern `inject()` pattern, and the new test coverage are all solid. There is one **high-severity bug** that must be fixed before merge, plus several lower-severity findings detailed in the inline comments.

---

### Summary of findings

| # | Area | Severity | File |
|---|------|----------|------|
| 1 | **Bug** – `Location.back()` loops back to the forbidden page when user was redirected by `roleGuard` | 🔴 High | `forbidden.component.ts:16` |
| 2 | **Bug** – `Location.back()` does nothing (or exits the app) when there is no prior history | 🟡 Medium | `forbidden.component.html:14` |
| 3 | **Semantics** – `<article>` is the wrong element for an error card | 🔵 Low | `forbidden.component.html:2` |
| 4 | **Angular Best Practices** – `RouterTestingModule` is deprecated in Angular 16+; project uses Angular 19 | 🔵 Low–Medium | `forbidden.component.spec.ts:3` |
| 5 | **UI/UX** – `64px` navbar height hardcoded in SCSS calculation instead of a token | 🔵 Low | `forbidden.component.scss:5` |
| 6 | **Tests** – Fixture not destroyed between `it()` blocks | 🔵 Low | `forbidden.component.spec.ts:25` |
| 7 | **Documentation** – `docs/context.md` item 36 describes the old component behaviour | 🔵 Low | `docs/context.md` |

---

### 1. Documentation

- **Problem:** `docs/context.md` item 36 still reads: *"A tela ForbiddenComponent exibe a mensagem de acesso negado com link de retorno ao dashboard"*. The page now has a rewritten message and a second action ("Voltar").
- **Why it matters:** `docs/context.md` is the authoritative source of truth for decisions and current state. Future contributors will read stale information.
- **Suggested improvement:** Update item 36 (and the "Estado Atual" narrative block at the bottom) to reflect the new two-action layout, the `Location.back()` behaviour, and the new message text.
- **Severity:** Low

---

### 2. Security

No issues found. The component renders no external data, uses no `innerHTML`/`bypassSecurityTrust*`, exposes no tokens, and has no console logs. ✅

---

### 3. Performance

No issues found. No subscriptions, no API calls, no complex template logic. `Location` is injected cleanly and the component has no memory-leak risk. ✅

---

### 4. API Integration

Not applicable — this component makes no HTTP calls. ✅

---

### 5. Code Quality

Generally good. `private readonly location = inject(Location)`, explicit `void` return type, Portuguese method name consistent with project conventions, and design tokens used throughout the SCSS. The nested `&:hover` was correctly replaced with a nested `.btn` rule inside the media query. ✅

---

### What's working well ✅

- `aria-labelledby="forbidden-title"` + `aria-hidden="true"` on the decorative number — correct and thoughtful.
- Responsive layout with `flex-wrap` + column stack on mobile.
- `clamp()` for fluid `font-size` on the error code.
- `inject()` over constructor injection — aligned with modern Angular.
- All new behaviour covered by dedicated tests.
- Development build confirmed working per PR description.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by FabioCarlesso — COMMENTED** [review]
# Review — PR #82: feat(auth): aprimora página 403

Revisão considerando o escopo da PR (página 403) e o commit adicional `24d2f34` (fix dropdown perfil admin).

---

## 1. Documentação

- **Problem:** `docs/context.md` (linhas 50 e 132) descreve a tela 403 como "exibe a mensagem de acesso negado com **link** de retorno ao dashboard" (singular). A PR adiciona um botão secundário `Voltar` (chamando `Location.back()`), de modo que agora há ação primária + secundária. O texto da doc ficou levemente desatualizado.
- **Why it matters:** O `context.md` é uma referência viva do comportamento; quem ler a doc vai ter expectativa diferente da implementação.
- **Suggested improvement:** Atualizar para algo como: "exibe código 403, mensagem orientativa e duas ações: link primário para o início e botão secundário 'Voltar' (`Location.back()`)".
- **Severity:** Low

---

## 2. Bugs e Edge Cases

### 2.1 `voltar()` sem fallback quando não há histórico
- **Problem:** `forbidden.component.ts:16` chama `this.location.back()` sem verificar se há entrada anterior no histórico. Quando o usuário abre `/403` em aba nova, recarrega a página, ou chega via link compartilhado, `history.back()` sai da SPA (volta para `about:blank`/página anterior do navegador) ou simplesmente não faz nada.
- **Why it matters:** Em um caso comum de "copiar URL e abrir em outra aba", o botão "Voltar" se torna armadilha — o usuário sai do app inteiro. Acontece especialmente em ambientes corporativos onde admins compartilham URLs.
- **Suggested improvement:**
  ```ts
  voltar(): void {
    if (window.history.length > 1) {
      this.location.back();
    } else {
      this.router.navigateByUrl('/');
    }
  }
  ```
  (Injetar `Router` e tratar o fallback). Adicionar teste cobrindo o caminho sem histórico.
- **Severity:** Medium

### 2.2 Inconsistência de padrão de "Voltar"
- **Problem:** Em todo o resto do app (`pacientes`, `planos`, `aulas`, `pagamentos`, `relatorios`, `admin/usuarios`), o botão `Voltar` é sempre um `routerLink` para um destino conhecido. Esta PR introduz pela primeira vez `Location.back()`.
- **Why it matters:** Comportamento imprevisível — clicar "Voltar" pode levar a páginas diferentes dependendo da origem; quebra a expectativa estabelecida pelo restante da UI.
- **Suggested improvement:** Considerar trocar a ação secundária por algo determinístico, como `routerLink="/login"` ("Sair da sessão") quando o usuário não tem perfil válido, ou simplesmente remover o secundário (a ação primária "Ir para o início" já cobre o caso). Se mantiver `Location.back()`, ao menos resolver 2.1.
- **Severity:** Low

---

## 3. Segurança

Sem problemas. Conteúdo é texto estático, sem `[innerHTML]`, sem dados sensíveis em log/console, sem bypass de guard. A rota `/403` continua pública (correto — é exibida quando o usuário autenticado tenta acessar área não permitida).

---

## 4. Performance

Sem problemas. Sem subscriptions, sem cálculos pesados, sem mudanças no bundle relevantes. `clamp(4.5rem, 16vw, 7.5rem)` é eficiente.

---

## 5. Angular Best Practices

- **Positivo:** uso de `inject()` em vez de constructor injection (consistente com o estilo já adotado em `usuario-form.component.ts`); standalone component; `OnPush` não é necessário aqui (componente sem inputs/observables).
- **Observação (Low):** O teste usa `RouterTestingModule`, deprecado desde Angular 15+ em favor de `provideRouter([])`. Não é regressão (já era o padrão da spec antes), mas para componentes novos/reescritos vale considerar a migração — alguns specs do projeto já usam `provideRouter([])` (ex.: `usuario-form.component.spec.ts`). **Severity: Low**.

---

## 6. API Integration

N/A para a página 403.

Sobre o commit adicional (`fix(admin/usuario-form)`): contrato alinhado ao `RoleResponseDTO` do backend (`{value, label}`); fallback para `ROLE_OPTIONS` em erro/lista vazia é mantido; tipagem do service corrigida; testes atualizados. **Looks good**.

---

## 7. UI/UX e Acessibilidade

### 7.1 `aria-label` em `<div>` sem role
- **Problem:** `forbidden.component.html:12` — `<div class="forbidden-actions" aria-label="...">`. ARIA recomenda usar `aria-label` apenas em elementos com role implícito ou explícito; em um `<div>` genérico, alguns screen readers ignoram o rótulo.
- **Why it matters:** O agrupamento das ações pode não ser anunciado corretamente para usuários de leitor de tela.
- **Suggested improvement:** Adicionar `role="group"` ao div, ou trocar por `<nav aria-label="...">` (como é "navegar para fora da página", `<nav>` cabe).
- **Severity:** Low

### 7.2 Redundância visual eyebrow + número grande
- **Problem:** O eyebrow "Erro 403" e o número grande `403` exibem a mesma informação visualmente. Para AT está OK (`aria-hidden="true"` no número grande). Para usuários sighted, é redundância.
- **Why it matters:** Mais ruído visual sem ganho informativo. Em telas pequenas, ocupa espaço desnecessário.
- **Suggested improvement:** Remover o eyebrow ou remover o número grande — manter apenas um. Sugiro manter o número grande (mais característico de páginas de erro) e remover o eyebrow.
- **Severity:** Low

### 7.3 Sem regressão de UX
Layout responsivo com breakpoint em 520px (botões empilhados em mobile) está adequado. Tokens do Design System aplicados corretamente (todas as variáveis usadas existem em `_tokens.scss`).

---

## 8. Tests

- **Cobertura presente:** título, mensagem, link primário, clique do botão secundário invocando `Location.back()`. **Looks good** para o caminho feliz.
- **Faltando:**
  - Edge case: comportamento quando `window.history.length === 1` (relacionado ao bug 2.1).
  - Verificação de acessibilidade: presença do `aria-labelledby`/`id="forbidden-title"` (snapshot ou `aria-*` checks).
- **Severity:** Low

---

## 9. Code Quality

- Nomenclatura de classes CSS alinhada (`forbidden-*` BEM-like).
- Templates legíveis, sem lógica.
- Estrutura de pastas mantida.
- **Looks good**, exceto pelas observações de UX/A11y acima.

---

## Resumo

| Severidade | Quantidade |
|------------|-----------|
| High       | 0          |
| Medium     | 1 (`Location.back()` sem fallback) |
| Low        | 5 (doc desatualizada, padrão Voltar inconsistente, `aria-label` em div, eyebrow redundante, falta teste edge case + RouterTestingModule deprecado) |

A PR está funcionalmente correta e visualmente alinhada ao Design System; o único ponto de atenção real é o `Location.back()` sem fallback, que pode levar o usuário para fora da SPA em deep links. Os demais são polimentos de qualidade.

**Comment by FabioCarlesso** [comment]
## Correções do code review aplicadas (`58cdd86`)

### Medium
- **`Location.back()` sem fallback** (`forbidden.component.ts:16`): `voltar()` agora verifica `window.history.length` e cai para `router.navigateByUrl('/')` quando não há histórico, evitando que o usuário saia da SPA em deep links abertos em aba nova / após refresh em `/403`.

### Low
- **Doc desatualizada**: `docs/context.md` (itens 36 e seção "Autorização por role") agora descreve as duas ações da tela 403 e o fallback do botão Voltar.
- **Padrão `Voltar` inconsistente**: mantido `Location.back()` (mais natural para uma página de erro genérica), mas com fallback determinístico para `/`, eliminando o risco de comportamento imprevisível do item Medium.
- **`aria-label` em `<div>` sem role**: adicionado `role="group"` ao container `.forbidden-actions` para garantir que o rótulo seja anunciado por screen readers.
- **Eyebrow "Erro 403" redundante**: removido — fica apenas o número grande, mais característico de páginas de erro. O título `Acesso negado` continua sendo o accessible name da seção via `aria-labelledby`.
- **Testes**: spec migrada de `RouterTestingModule` (deprecado) para `provideRouter([])`. Adicionados três testes:
  - `should fall back to navigating home when there is no previous history`
  - `should expose the action group with role="group" and an accessible label`
  - `should expose the title as the section accessible name`

### Verificação
- `npx tsc --noEmit -p tsconfig.spec.json` ✅ sem erros
- `npx ng test --include='src/app/pages/auth/forbidden/**/*.spec.ts'` ✅ 8/8 testes passando
